src/tool.js

```diff
@@ -0,0 +1,33 @@
+exports.buildForm = tools => {
+  let copy = "";
+  try {
+    tools.forEach(tool => {
+      if (tool.type) {
+        switch (tool.type) {
+          case "text":
+          case "username":
```

Comment:
Your form builder seems to be narrowly designed to only fulfill the direct requirements of the assignment. The idea behind a tool is that it is reusable, however. This form builder tool should be able to be reused to build any simple form such as a contact form, or even an Ecommerce checkout form.